pkg/registry/apis/provisioning/files.go

```diff
@@ -200,7 +203,27 @@ func (s *filesConnector) Connect(ctx context.Context, name string, opts runtime.
 }
 
 // listFolderFiles returns a list of files in a folder
-func (s *filesConnector) listFolderFiles(ctx context.Context, filePath string, ref string, readWriter repository.ReaderWriter) (*provisioning.FileList, error) {
+func (c *filesConnector) listFolderFiles(ctx context.Context, filePath string, ref string, readWriter repository.ReaderWriter) (*provisioning.FileList, error) {
+	auth, ok := authlib.AuthInfoFrom(ctx)
+	if !ok {
+		return nil, fmt.Errorf("missing auth info in context")
+	}
+
+	// Listing files requires permission to list dashboards
+	check, err := c.access.Check(ctx, auth, authlib.CheckRequest{
+		Group:     resources.DashboardResource.Group,
+		Resource:  resources.DashboardResource.Resource,
+		Namespace: readWriter.Config().Namespace,
+		Verb:      utils.VerbList,
+	})
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, fmt.Errorf("check dashboard access: %w", err)
```

Reply:
🤔 do you know how to keep the 403 with this?  seems we lose the useful info by wrapping here

Reply:
I would expect the responder to to be able to unwrap the apierrors when using `responder.Error(err)`. If that's not the case, maybe we should create a wrapper around it to this for all the connectors as we throw these apierrors very deep in the code.